src/middlewares/checkParams.js

```diff
@@ -0,0 +1,32 @@
+import Models from '../db/models';
+import errorResponse from '../helpers/index';
+
+const { Bookmark } = Models;
+
+
+const paramsChecker = {
+  idChecker: (request, response, next) => {
+    const { id } = request.params;
+    const validId = /^[0-9]+$/;
```

Comment:
What happens when id is zero or starts with a zero?

Reply:
Oh, thank you for spotting that out. Well, as it stands, when Id is zero, the middleware does not terminate the process. I will make adjustments as soon as possible

Reply:
I think you should terminate the process if `id` is zero, and either terminate or strip off zero if the first number in the `id` is a zero (e.g `09` should be terminated or seen as a `9`)

Reply:
I understand your point completely, but all the middleware does is to  ensure the id is a number. If you entered zero, you get a response that its reference does not exist. I have other checks in place for that. Got it?

Reply:
@frostyblok I was about to agree with @marcdomain, but I think your explanation covers it 👍